cmd/server/main.go

```diff
@@ -26,6 +33,47 @@ func main() {
 
 		userRepo = mssqldb.NewUserRepo(db)
 		todoRepo = mssqldb.NewTodoRepo(db)
+	case "mongodb":
+		atlas_uri := config.GetAtlasUri()
+		client, err := mongo.NewClient(options.Client().ApplyURI(atlas_uri))
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+		defer cancel()
+
+		err = client.Connect(ctx)
+		if err != nil {
+			log.Fatal(err)
+		}
+		defer client.Disconnect(ctx)
+
+		database := client.Database("todoapp")
+
+		userCollection := database.Collection("users")
+		_, err = userCollection.Indexes().CreateOne(
+			context.Background(),
+			mongo.IndexModel{
+				Keys:    bson.D{{Key: "username", Value: 1}},
+				Options: options.Index().SetUnique(true),
+			},
+		)
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		todoCollection := database.Collection("todos")
+		_, err = todoCollection.Indexes().CreateOne(
+			context.Background(),
+			mongo.IndexModel{
+				Keys:    bson.D{{Key: "id", Value: 1}},
+				Options: options.Index().SetUnique(true),
```

Comment:
do not create indexes every time we start up server, mongo will have to redo indexing, for example, we have 1 million data row, then it has to re-indexing those 1 million records every time we run this server?